fix: resolve SonarCloud S7636 quality gate blockers in workflow files

## Problem

SonarCloud quality gate is in **ERROR** state due to 3 S7636 (secret exposure in workflow) findings. These are the only blockers preventing the gate from passing.

## Fixes Required

### 1. `sync-wiki.yml` — Line 53 (token in git URL)

Move `GITHUB_TOKEN` out of the git remote URL into an env var:

```yaml
      - name: Sync wiki content
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          # ... (all existing rsync/copy steps unchanged) ...
          cd wiki
          git add .
          if git diff --staged --quiet; then
            echo "No changes to commit"
            exit 0
          fi
          git commit -m "Sync wiki from source repository"
          git remote set-url origin "https://x-access-token:${GH_TOKEN}@github.com/${{ github.repository }}.wiki.git"
          git push origin
```

### 2. `sonarcloud.yml` — Line 43 (secret in `if` condition)

```yaml
      - name: Check if SonarCloud token is set
        id: check_token
        env:
          SONARCLOUD_GITHUB: ${{ secrets.SONARCLOUD_GITHUB }}
        run: |
          if [ -z "${SONARCLOUD_GITHUB}" ]; then
```

### 3. `code-quality.yml` — Line 147 (secret in `if` condition)

```yaml
      - name: Check if Codacy token is set
        id: check_codacy_token
        env:
          CODACY_PROJECT_TOKEN: ${{ secrets.CODACY_PROJECT_TOKEN }}
        run: |
          if [ -z "${CODACY_PROJECT_TOKEN}" ]; then
```

### 4. `bin/playground-test.sh` — SC2155 (6 instances)

Split `local var=$(cmd)` into two lines for all 6 occurrences in `check_cli()`, `start_playground()`, `stop_playground()`, and `check_status()`.

### 5. `.shellcheckrc` — Suppress vendor SC2223

Create `.shellcheckrc` at repo root to exclude vendor directory from ShellCheck.

## Acceptance Criteria

- [ ] SonarCloud quality gate passes (no ERROR state)
- [ ] ShellCheck reports 0 first-party warnings
- [ ] All 3 S7636 findings resolved at source
- [ ] PR passes CI

## Source

Identified by CodeRabbit analysis on quality review issue #92 (2026-03-20).

--- a/.shellcheckrc
+++ b/.shellcheckrc
@@ -3,4 +3,4 @@
 
 # Exclude vendor and third-party directories from ShellCheck analysis.
 # These directories contain code we do not maintain and should not be linted.
-exclude=vendor,node_modules
+exclude=vendor/,node_modules/

